backend/app/storage.py

```diff
@@ -160,3 +160,14 @@ async def put_thread(
             "name": name,
             "updated_at": updated_at,
         }
+
+
+async def get_or_create_user(sub: str) -> tuple[User, bool]:
```

Comment:
`put_*` function name implies that a resource can be edited. Since we want to keep the user object as minimal as possible with `user_id` and `sub` attributes, none of which make sense to be editable, I decided to not have a put_user function. Since user retrieval/creation only happens in auth, I decided to go with get_or_create_user. I can get rid of the boolean element of the tuple if you want (I can also use a put_user method if you think that's best).

Reply:
My opinion: I feel like it's too premature to suggest that the `user` object should/may never change. For example, supporting a `name` field in the future seems like a reasonable feature. Now that I'm thinking about it, it may be wise to include the `updated_at` field as well.

@nfcampos, thoughts?

Reply:
I agree adding a created_at timestamp at least would be useful

Reply:
Sounds good. Added created_at. Let's add more fields later, when we see their specific use cases in the project.